src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -107,6 +107,15 @@ export default class DefaultMessagingSession implements MessagingSession {
   private async startConnecting(reconnecting: boolean): Promise<void> {
     this.isConnecting = true;
     try {
+      if (!reconnecting) {
+        this.reconnectController.reset();
+      }
+      if (this.reconnectController.hasStartedConnectionAttempt()) {
+        this.reconnectController.startedConnectionAttempt(false);
+      } else {
+        this.reconnectController.startedConnectionAttempt(true);
+      }
+
       // reconnect needs to re-resolve endpoint url, which will also refresh credentials on client if they are expired.
       let endpointUrl = !reconnecting ? this.configuration.endpointUrl : undefined;
       if (endpointUrl === undefined) {
```

Comment:
I think you also have to handle the error here, sorry I missed that there is no `catch` here:
https://github.com/aws/amazon-chime-sdk-js/pull/2179#discussion_r860332917

Reply:
The `getMessagingSessionEndpoint()` can error out here as well.

Reply:
There is a finally block around this whole thing to reset the boolean.  Not sure what if anything we should do with the caught exception.  Thoughts?